gbdxtools/auth.py

```diff
@@ -42,9 +42,39 @@ def __init__(self, **kwargs):
         except Exception as err:
             print(err)
 
+        def expire_token(r, *args, **kwargs):
+            """
+            Requests a new token if 401, retries request, mainly for auth v2 migration
+            :param r:
+            :param args:
+            :param kwargs:
+            :return:
+            """
+            if r.status_code == 401:
+                try:
+                    # remove hooks so it doesn't get into infinite loop
+                    r.request.hooks = None
+                    # expire the token
+                    gbdx_auth.expire_token(token_to_expire=self.gbdx_connection.token,
+                                           config_file=kwargs.get('config_file'))
```

Comment:
@Brett55 Sometimes, like in GBDX Notebooks and Tasks, we dont have a config file for auth. We use ENV Vars, will this fail if there's no config file used?  Also since this is a hook, what determines the args and kwargs that get passed to this?

Reply:
same here, need to default the get to None: `kwargs.get('config_file')`

Reply:
None is returned implicitly in that .get method

Reply:
@Brett55 ahh you're right, i've had it choke before when testing gets without a default but i actually think that a bit a of diff scenario